contrib/rabbitmq/samples/README.md

```diff
@@ -0,0 +1,141 @@
+    customresourcedefinition.apiextensions.k8s.io/rabbitmqsources.sources.eventing.knative.dev configured
+    service/rabbitmq-controller created
+    statefulset.apps/rabbitmq-controller-manager created
+    ```
+2. Check that the `rabbitmq-controller-manager-0` pod is running.
+    ```
+    $ kubectl get pods -n knative-sources
+    NAME                         READY     STATUS    RESTARTS   AGE
+    rabbitmq-controller-manager-0   1/1       Running   0          42m
+    ```
+3. Check the `controller-manager-0` pod logs.
+    ```
+    $ kubectl logs rabbitmq-controller-manager-0 -n knative-sources
+    2019/03/19 22:25:54 Registering Components.
+    2019/03/19 22:25:54 Setting up Controller.
+    2019/03/19 22:25:54 Adding the Rabbitmq Source controller.
+    2019/03/19 22:25:54 Starting Rabbitmq controller.
+    ```
+    
+#### Rabbitmq Event Source
+1. Modify `contrib/rabbitmq/samples/event-source.yaml` accordingly with brokers, topic, routing key etc... 
+2. Build and deploy the event source.
+    ```
+    $ ko apply -f contrib/rabbitmq/samples/event-source.yaml
+    ...
+    rabbitmqsource.sources.eventing.knative.dev/rabbitmq-source created
+    ```
+3. Check that the event source pod is running. The pod name will be prefixed with `rabbitmq-source`.
+    ```
+    $ kubectl get pods
+    NAME                                  READY     STATUS    RESTARTS   AGE
+    rabbitmq-source-xlnhq-5544766765-dnl5s   1/1       Running   0          40m
+    ```
+4.  Ensure the Rabbitmq Event Source started with the necessary configuration.
+    ```
+    $ kubectl logs rabbitmq-source-xlnhq-5544766765-dnl5s
+    {"level":"info","ts":"2019-03-19T22:31:52.689Z","caller":"receive_adapter/main.go:97","msg":"Starting Rabbitmq Receive Adapter...","adapter":{"Brokers":"...","topic":"...","routingKey":"...","Net":{"SASL":{"Enable":true,"User":"...","Password":"..."},"TLS":{"Enable":true}},"SinkURI":"http://event-display.default.svc.cluster.local/"}}
+    ```
+    
+#### Event Display 
+1. Build and deploy the Event Display Service.
+    ```
+    $ ko apply -f contrib/rabbitmq/samples/event-display.yaml
+    ...
+    service.serving.knative.dev/event-display created
+    ``` 
+2. Ensure that the Service pod is running. The pod name will be prefixed with `event-display`.
+    ```
+    $ kubectl get pods
+    NAME                                            READY     STATUS    RESTARTS   AGE
+    event-display-00001-deployment-5d5df6c7-gv2j4   2/2       Running   0          72s
+    ...
+    ```   
+    
+### Verify
+1. Produce the message shown below to Rabbitmq.
+    ```
+    "Hello World"
+    ```
+2. Check that the Rabbitmq Event Source consumed the message and sent it to its sink properly.
+    ```
+    $ kubectl logs rabbitmq-source-xlnhq-5544766765-dnl5s
+    ...
+    {"level":"info","ts":1554244010.3225584,"logger":"fallback","caller":"adapter/adapter.go:158","msg":"Received: {value 15 0 Hello World <nil>}"}
```

Comment:
How do you pass context in the cloud event? How did there be `nil` there?

Reply:
Its the format in which zap serializes & prints the log.
```
zap.Any("value", string(msg.Body))
```

Reply:
Is the because msg.Body is a []data ?